app/models/concerns/geo_concerns/metadata_extraction_helper.rb

```diff
@@ -2,16 +2,16 @@ module GeoConcerns
   module MetadataExtractionHelper
     # Extracts properties from the constitutent external metadata file
     # @return [Hash]
-    def extract_metadata
+    def extract_metadata(id)
       return {} if metadata_files.blank?
-      # TODO: Does not support multiple external metadata files
-      raise NotImplementedError if metadata_files.length > 1
-      metadata_files.first.extract_metadata
+      metadata_files.each do |metadata_file|
```

Comment:
This requires that there is some implicit ordering in the metadata files being structured within the AJAX request .  While I think this is a trivial (or even pedantic) point, my only concern might be for cases where the UI might be deprecated for something more modular (i. e. RESTful API support).  Should it be desirable that we address those cases, I can work to provide some documentation specifying that this ordering determines precisely which attributes are overridden during the population.